python_on_whales/components/pod/cli_wrapper.py

```diff
@@ -0,0 +1,778 @@
+                This function will then return an iterator that will yield a
+                tuple `(source, content)` with `source` being `"stderr"` or
+                `"stdout"`. `content` is the content of the line as bytes.
+                Take a look at [the user guide](https://gabrieldemarmiesse.github.io/python-on-whales/user_guide/docker_run/#stream-the-output)
+                to have an example of the output.
+
+        # Returns
+            `str` if `stream=False` (the default), `Iterable[Tuple[str, bytes]]`
+            if `stream=True`.
+
+        # Raises
+            `python_on_whales.exceptions.NoSuchPod` if the container does not exist.
+
+        If you are a bit confused about `follow` and `stream`, here are some use cases.
+
+        * If you want to have the logs up to this point as a `str`, don't use those args.
+        * If you want to stream the output in real time, use `follow=True, stream=True`
+        * If you want the logs up to this point, but you don't want to fit all the logs
+        in memory because they are too big, use `stream=True`.
+        """
+
+        # first we verify that the pod exists and raise an exception if not.
+        self.inspect(pod)
+
+        full_cmd = self.docker_cmd + ["pod", "logs"]
+        full_cmd.add_simple_arg("--container", container)
+        full_cmd.add_flag("--names", names)
+        full_cmd.add_simple_arg("--since", format_time_arg(since))
+        full_cmd.add_simple_arg("--tail", tail)
+        full_cmd.add_flag("--timestamps", timestamps)
+        full_cmd.add_simple_arg("--until", format_time_arg(until))
+        full_cmd.add_flag("--follow", follow)
+        full_cmd.append(pod)
+
+        iterator = stream_stdout_and_stderr(full_cmd)
+
+        if stream:
+            return iterator
+        else:
+            return "".join(x[1].decode() for x in iterator)
+
+    def pause(
+        self,
+        x: Union[ValidPod, Sequence[ValidPod]] = (),
+        /,
+        *,
+        all: bool = False,
+        latest: bool = False,
+    ) -> None:
+        """Pauses one or more pods
+
+        Parameters:
+            x: One or more pods to pause
+            all: Pause all pods
+            latest: Pause the latest pod
+
+        # Raises
+            `python_on_whales.exceptions.NoSuchPod` if any pods does not exist.
+        """
+        pods = to_list(x)
+        if len(pods) == 0 and not latest and not all:
+            return
+
+        full_cmd = self.docker_cmd + ["pod", "pause"]
+        full_cmd.add_flag("--all", all)
+        full_cmd.add_flag("--latest", latest)
+        full_cmd.extend([str(p) for p in pods])
+
+        run(full_cmd)
+
+    def prune(self) -> None:
+        """Remove pods that are not running."""
+        full_cmd = self.docker_cmd + ["pod", "prune", "--force"]
+        run(full_cmd)
+
+    ps = list
+
+    def remove(
+        self,
+        x: Union[ValidPod, Sequence[ValidPod]] = (),
+        /,
+        *,
+        all: bool = False,
```

Comment:
I wonder if there is an elegant way to say that `all` can't be used at the same time as `x`. Maybe with overloads? Or maybe with a special sentinel value? I'm always in favor of having the IDE help as much as possible. We also might want to verify that both all and x are not set at the same time and raise an error if this is the case.

Reply:
I'll have a go at using overloads - might be a bit verbose but I can see it should be helpful. A sentinel value might be nice (could just have a global `ALL = object()` and `LATEST = object()`) but perhaps a bit less obvious than just mirroring the CLI. Probably best to have a separate issue for that if you want to track the idea.

Reply:
I've removed support for `all` and `latest` in this PR since that's what a lot of the comments are about, so this can be discussed and addressed separately (https://github.com/gabrieldemarmiesse/python-on-whales/issues/588). E.g. there's also equivalent flags on some `podman container` commands.